lxd/db/cluster/identities.go

```diff
@@ -404,8 +404,8 @@ func (i *Identity) ToAPI(ctx context.Context, tx *sql.Tx, canViewGroup auth.Perm
 }
 
 // ActivateTLSIdentity updates a TLS identity to make it valid by adding the fingerprint, PEM encoded certificate, and setting
-// the type to api.IdentityTypeCertificateClient.
-func ActivateTLSIdentity(ctx context.Context, tx *sql.Tx, identifier uuid.UUID, cert *x509.Certificate) error {
+// the type.
+func ActivateTLSIdentity(ctx context.Context, tx *sql.Tx, identifier uuid.UUID, cert *x509.Certificate, identityType IdentityType) error {
```

Comment:
The idea behind adding the `Activate` or `ActiveType` function here was so that the identity type doesn't need to be passed in.

Can we get the identity from the database by its identifier (not just it's ID) and then call `ActiveType` on the identity type that the returned identity gives us?